# Unemployment Analysis with Python

## Project Overview

This project analyzes unemployment data in India to identify regional and
temporal trends in unemployment rates, with a focus on the impact of the
COVID-19 pandemic.

## Objectives

- Explore and understand the unemployment dataset
- Analyze unemployment rates across different regions
- Study monthly and time-based unemployment trends
- Compare unemployment before and after COVID-19
- Visualize important patterns using charts
- Identify regions with higher average unemployment rates

## Technologies Used

- Python
- Pandas
- NumPy
- Matplotlib
- Seaborn
- Jupyter Notebook

## Dataset

The dataset contains unemployment-related information for different regions
of India, including unemployment rate, employment rate, and labour
participation rate.

## 1. Import Required Libraries

The following Python libraries are used in this project:

- **Pandas:** Used for loading, cleaning, and analyzing the unemployment dataset.
- **NumPy:** Used for numerical operations and calculations.
- **Matplotlib:** Used for creating data visualizations and charts.
- **Seaborn:** Used for statistical visualizations such as heatmaps and plots.

These libraries provide the basic tools required for exploratory data analysis and visualization.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 2. Load the Dataset

The unemployment dataset is stored in a CSV (Comma-Separated Values) file.

We use the Pandas `read_csv()` function to load the dataset into a DataFrame.

A DataFrame is a table-like structure in Pandas that allows us to easily
inspect, clean, analyze, and visualize the data.

In [6]:
df = pd.read_csv("Unemployment_Rate_upto_11_2020.csv")

df.head()

,Region,Date,Frequency,Estimated Unemployment Rate (%),Estimated Employed,Estimated Labour Participation Rate (%),Region.1,longitude,latitude
0,Andhra Pradesh,31-01-2020,M,5.48,16635535,41.02,South,15.9129,79.74
1,Andhra Pradesh,29-02-2020,M,5.83,16545652,40.90,South,15.9129,79.74
2,Andhra Pradesh,31-03-2020,M,5.79,15881197,39.18,South,15.9129,79.74
3,Andhra Pradesh,30-04-2020,M,20.51,11336911,33.10,South,15.9129,79.74
4,Andhra Pradesh,31-05-2020,M,17.43,12988845,36.46,South,15.9129,79.74


## 3. Dataset Shape

The `shape` attribute is used to find the number of rows and columns in the dataset.

It returns a tuple in the format:

`(number of rows, number of columns)`

This helps us understand the size of the dataset before performing further analysis.

In [10]:
df.shape

(267, 9)

### Dataset Size

The dataset contains **267 rows and 9 columns**.

- **Rows:** 267
- **Columns:** 9

This gives us an initial understanding of the size of the dataset before performing data cleaning and exploratory analysis.

## 4. Dataset Information

The `info()` method provides important information about the dataset, including:

- Column names
- Number of non-null values
- Data types
- Memory usage

This helps us identify columns that may require data cleaning or type conversion.

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 267 entries, 0 to 266
Data columns (total 9 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Region                                    267 non-null    object 
 1    Date                                     267 non-null    object 
 2    Frequency                                267 non-null    object 
 3    Estimated Unemployment Rate (%)          267 non-null    float64
 4    Estimated Employed                       267 non-null    int64  
 5    Estimated Labour Participation Rate (%)  267 non-null    float64
 6   Region.1                                  267 non-null    object 
 7   longitude                                 267 non-null    float64
 8   latitude                                  267 non-null    float64
dtypes: float64(4), int64(1), object(4)
memory usage: 18.9+ KB


## 5. Checking for Missing Values

Missing values can affect data analysis and machine learning models.

We use the `isnull()` method to identify missing values in each column.

The `sum()` method counts the total number of missing values for every column.

This step helps us determine whether the dataset requires missing-value treatment before analysis.

In [13]:
df.isnull().sum()

Region                                      0
 Date                                       0
 Frequency                                  0
 Estimated Unemployment Rate (%)            0
 Estimated Employed                         0
 Estimated Labour Participation Rate (%)    0
Region.1                                    0
longitude                                   0
latitude                                    0
dtype: int64

### Missing Value Observation

The missing-value check shows that all columns contain **0 missing values**.

Therefore, no missing-value treatment is required for this dataset at this stage.
The dataset is complete with respect to missing values.

## 6. Cleaning Column Names

While inspecting the dataset, we noticed that some column names contain
unnecessary leading or trailing spaces.

Extra spaces in column names can cause errors when accessing columns.

We will remove these unwanted spaces using the `str.strip()` method so that
the column names are clean and easier to work with.

In [16]:
df.columns = df.columns.str.strip()

df.columns

Index(['Region', 'Date', 'Frequency', 'Estimated Unemployment Rate (%)',
       'Estimated Employed', 'Estimated Labour Participation Rate (%)',
       'Region.1', 'longitude', 'latitude'],
      dtype='object')

## 7. Convert Date Column

The `Date` column was initially stored as an object (text) data type.

For time-series analysis, we need the dates in a proper datetime format.
Therefore, we convert the `Date` column using Pandas `to_datetime()` function.

This will allow us to analyze unemployment trends by month and compare
different time periods.

In [18]:
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 267 entries, 0 to 266
Data columns (total 9 columns):
 #   Column                                   Non-Null Count  Dtype         
---  ------                                   --------------  -----         
 0   Region                                   267 non-null    object        
 1   Date                                     267 non-null    datetime64[ns]
 2   Frequency                                267 non-null    object        
 3   Estimated Unemployment Rate (%)          267 non-null    float64       
 4   Estimated Employed                       267 non-null    int64         
 5   Estimated Labour Participation Rate (%)  267 non-null    float64       
 6   Region.1                                 267 non-null    object        
 7   longitude                                267 non-null    float64       
 8   latitude                                 267 non-null    float64       
dtypes: datetime64[ns](1), float64(4), int64(1), 

### Date Conversion Observation

The `Date` column was successfully converted from an object data type to
`datetime64[ns]`.

The dataset uses a day-month-year date format, so `dayfirst=True` was used
during conversion.

This conversion is important because it allows us to perform time-series
analysis and compare unemployment trends across different periods.